### PR TITLE
Feature/447 implement get list of all local authorities

### PR DESCRIPTION
**Pull request by ian-wells-nimble**
**Context:**

When a user is creating a new temporary accommodation property, they need to be able to select a local authority area from a list. This list of course needs to be the actual names of the areas. When all of the data is passed to the server to create a new property, it will include the ID of the area which will be persisted.

This change exposes a new endpoint that the client can call to retrieve a full list of area objects with all of their attributes.

### CODE REVIEW

File: src/main/resources/static/api.yml [hunk 1]
```diff
@@ -1032,6 +1032,26 @@ paths:
           $ref: '#/components/responses/403Response'
         500:
           $ref: '#/components/responses/500Response'
+  /reference-data/local-authorities:
```
Comment (edavey) [inline src/main/resources/static/api.yml:1035]: I think maybe we ought to be consistent with the term "local authority **areas**"?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/ReferenceDataController.kt [hunk 2]
```diff
@@ -3,21 +3,9 @@ package uk.gov.justice.digital.hmpps.approvedpremisesapi.controller
...
-import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.LostBedReasonRepository
-import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.MoveOnCategoryRepository
-import uk.gov.justice.digital.hmpps.approvedpremisesapi.transformer.CancellationReasonTransformer
-import uk.gov.justice.digital.hmpps.approvedpremisesapi.transformer.DepartureReasonTransformer
-import uk.gov.justice.digital.hmpps.approvedpremisesapi.transformer.DestinationProviderTransformer
-import uk.gov.justice.digital.hmpps.approvedpremisesapi.transformer.LostBedReasonTransformer
-import uk.gov.justice.digital.hmpps.approvedpremisesapi.transformer.MoveOnCategoryTransformer
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.*
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.*
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.transformer.*
```
Comment (edavey) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/ReferenceDataController.kt]: I think I've seen elsewhere that using wildcard imports is not the done thing at HMPPS, but Joe will know much better than me?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/ReferenceDataController.kt [hunk 3]
```diff
@@ -3,9 +3,24 @@ package uk.gov.justice.digital.hmpps.approvedpremisesapi.controller
...
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.DestinationProviderRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.LocalAuthorityAreaRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.LostBedReasonRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.MoveOnCategoryRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.transformer.CancellationReasonTransformer
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.transformer.DepartureReasonTransformer
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.transformer.DestinationProviderTransformer
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.transformer.LocalAuthorityAreaTransformer
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.transformer.LostBedReasonTransformer
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.transformer.MoveOnCategoryTransformer
```
Comment (edavey) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/ReferenceDataController.kt:23]: ah, ok, I see you've gone back to individual imports here. Would you mind correcting in the previous commit? That way the history is clearer and presents an easy narrative -- as we are not squashing commits when we merge to main...

**Review by edavey — COMMENTED** [review]
👍 nice!

 just a couple of points on:

- naming of "local authority area"
- rewriting of commits to keep the commit narrative clear and allow commit-by-commit reviewing

**Comment by tahb** [comment]
If possible [I think we're ready](https://github.com/ministryofjustice/hmpps-approved-premises-api/pull/154#issuecomment-1282679489) to put this straight into `main`

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/ReferenceDataController.kt (hunk 2 above)
Comment (brownbearsoftware) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/ReferenceDataController.kt]: Indeed, in IntelliJ if you go `File`->`Settings`->`Editor`->`Code Style`->`Kotlin` and set it up like so (making sure to remove anything in the bit in red) you should be on ktLint's good side :)

image

**Comment by ian-wells-nimble** [comment]
@edavey @brownbearsoftware I have pushed up a revision

**Comment by ian-wells-nimble** [comment]
> If possible [I think we're ready](https://github.com/ministryofjustice/hmpps-approved-premises-api/pull/154#issuecomment-1282679489) to put this straight into `main`

Oh okay shall I edit this to merge straight in to main once approved?

**Comment by tahb** [comment]
@ian-wells-nimble I'd suggest that we get https://github.com/ministryofjustice/hmpps-approved-premises-api/pull/154 into `main` first as you mentioned there would should be some conflicts. Then we can change the target branch of this pull request to `main`.

**Review by brownbearsoftware — APPROVED** [review]

**Pull request merged**